app/content/profile/index.js

```diff
@@ -230,55 +232,68 @@ function Index() {
+      showDialog(
+        "No Change Detected",
+        "The new name must be different from the current name.",
+      );
+      return;
+    }
+    if (passwordInputVisible && !(currentPassword && newPassword)) {
+      showDialog("Incomplete Form", "Please fill out all the password fields.");
+      return;
+    }
+    if (newPassword !== newPasswordCheck) {
+      showDialog(
+        "Passwords Do Not Match",
+        "The new passwords you entered do not match. Please try again.",
+      );
+      return;
+    }
+    try {
+      if (passwordInputVisible) {
         // attempt updating the password
-        try {
-          // re-authenticate the user and check if the provided current password is valid
-          const userCredential = await signInWithEmailAndPassword(
-            auth,
-            userEmail,
-            currentPassword,
-          );
-
-          // once re-authenticated, update the password
-          updatePassword(userCredential.user, newPassword)
-            .then(() => {
-              // Update successful
-              setCurrentPassword(""); // Clear password fields
-              setNewPassword("");
-              bottomSheetModalRef.current.close();
-              setPasswordInputVisible(false);
-              setSnackbarMessage("Password updated successfully");
-              setSnackbarVisible(true); // Show success snackbar
-            })
-            .catch((error) => {
-              // Update failed
-              console.log("password update error:", error.message);
-              showDialog(
-                "New password is too short",
-                "Provided new password must be at least 6 characters long!",
-              );
-            });
-        } catch (e) {
-          showDialog("Error", e.message);
-          console.log(e.message);
-        }
+        // re-authenticate the user and check if the provided current password is valid
+        const userCredential = await signInWithEmailAndPassword(
+          auth,
+          userEmail,
+          currentPassword,
+        );
+
+        // once re-authenticated, update the password
+        await updatePassword(userCredential.user, newPassword);
+
+        // Update successful
+        resetForm();
+        bottomSheetModalRef.current.close();
+        setPasswordInputVisible(false);
+        showSnackBar("Password updated successfully");
       }
+      //doing password first because it checks if the user's entered password is correct, so the form is submitted at once
+      if (newName !== userData.name) {
+        // check if they request to update their name to a new one
+        await updateDoc(doc(db, "teams", currentTeamId, "users", userId), {
+          name: newName,
+        });
+
+        // invalidate cache on successful name update
+        invalidateMultipleKeys(queryClient, [["userInfo"]]);
+        bottomSheetModalRef.current.close();
+        showSnackBar("Name field updated successfully");
```

Comment:
While slightly irrelevant to this PR's changes, I wanted to note that the snack bar has a weird positioning gap on iPhone15 Pro:

<img width="461" alt="Screenshot 2024-05-24 at 3 31 53 PM" src="https://github.com/Golf-Drill-Challenge-App/Golf-App/assets/91341345/c978919c-792c-4512-bb46-26439266863c">